components/ai_rewriter/resources/page/components/Layout.tsx

```diff
@@ -75,38 +65,41 @@ const Root = styled.div`
 `
 
 export default function Layout() {
-  const { undo, canUndo, redo, canRedo, erase, canErase, generate, close, openSettings } = useRewriterContext()
+  const context = useRewriterContext()
   return <Root>
     <TitleRow>
       <Icon name='product-brave-leo' />
       <TitleText>leo writer</TitleText>
       <Label color='blue'>PREMIUM</Label>
       <DialogControls>
-        <Button fab kind='plain-faint' size='small' onClick={openSettings}>
+        <Button fab kind='plain-faint' size='small' onClick={context.openSettings}>
           <Icon name='settings' />
         </Button>
-        <Button fab kind='plain-faint' size='small' onClick={close}>
+        <Button fab kind='plain-faint' size='small' onClick={context.close}>
           <Icon name='close' />
         </Button>
       </DialogControls>
     </TitleRow>
-    <FiltersContainer>
-      <InitialText />
-    </FiltersContainer>
-    <NoContent />
+    {context.generatedText
+      ? <ModifyGeneration />
+      : <BeginGeneration />}
```

Comment:
Just out of curiosity, context.generatedText is working now? On my local env I wasn't able to see <ModifyGeneration> but undo button was enabled, which feels like they out of sync.

Reply:
I think that might mean there was an error that came back - I updated this to not push to history unless the new generated text is different to the old.